### ISSUE TITLE
Bug: @typescript-eslint/typescript-estree does not support TypeScript 5.8

### ISSUE THREAD

**Issue by martinszeltins** [typescript-eslint/typescript-eslint#10975]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### Repro Code

```TypeScript
It looks like typescript-estree does not support TypeScript 5.8 yet. I am using the latest version of typescript-eslint

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.


* @typescript-eslint/typescript-estree version: 8.24.0
* Supported TypeScript versions: >=4.8.4 <5.8.0
* Your TypeScript version: 5.8.2


Please only submit bug reports when using the officially supported version.
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/<rule-name>": ["error", ...<options>],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Should work

### Actual Result

Not working

### Versions

{
  "devDependencies": {
    "@eslint/eslintrc": "3.3.0",
    "@eslint/js": "9.22.0",
    "@typescript-eslint/eslint-plugin": "8.27.0",
    "@typescript-eslint/parser": "8.27.0",
    "@vue/eslint-config-typescript": "14.5.0",
    "eslint": "9.22.0",
    "eslint-plugin-vue": "10.0.0",
    "typescript-eslint": "8.27.0",
    "globals": "16.0.0",
    "typescript": "5.8.2",
    "vue-eslint-parser": "10.1.1",
    "vue-tsc": "2.2.0"
  },
}

**Comment by kirkwaiblinger** [comment]
Hi @martinszeltins !

The error message indicates that you have typescript-estree at version 8.24.0, but TS 5.8 support was added in 8.26 0, so this is expected (see https://github.com/typescript-eslint/typescript-eslint/releases/tag/v8.26.0).

What seems to be happening is that you have a different version of typescript-estree installed than you expect. I'd recommend checking `yarn why` or your package manager's equivalent to debug the unexpected dependency version.

**Comment by martinszeltins** [comment]
> Hi [@martinszeltins](https://github.com/martinszeltins) !
> 
> The error message indicates that you have typescript-estree at version 8.24.0, but TS 5.8 support was added in 8.26 0, so this is expected (see [`v8.26.0` (release)](https://github.com/typescript-eslint/typescript-eslint/releases/tag/v8.26.0)).
> 
> What seems to be happening is that you have a different version of typescript-estree installed than you expect. I'd recommend checking `yarn why` or your package manager's equivalent to debug the unexpected dependency version.

Ah, I see. I did not directly install `typescript-estree` myself. It must be installed as dep of another package. I suppose in this case I can just directly install it myself and specify `8.26.0` version. Thanks @kirkwaiblinger!

**Comment by bradzacher** [comment]
You can use your package manager's "resolutions" or "overrides" feature to ensure your version is what you expect.

But either way - we support 5.8 - the issue is your local versions.
